### DISCUSSION TITLE
How to specify different `packageName` for multiple packages under one repository (Homebrew)

### DISCUSSION THREAD

**Discussion by sgoudham** [renovatebot/renovate#29237] [Request Help]
### What would you like help with?

I would like help with my configuration

### How are you running Renovate?

Mend Renovate hosted app on github.com

### If you're self-hosting Renovate, tell us which platform (GitHub, GitLab, etc) and which version of Renovate.

GitHub App (`v37.368.10`)

### Please tell us more about your question or problem

Some prior context can be found in https://github.com/renovatebot/renovate/discussions/29175.

My overall goal is that I want to automate package updates from [catppuccin-toolbox](https://github.com/catppuccin/toolbox) within [catppuccin/homebrew-tap](https://github.com/homebrew-tap).

I have multiple custom tags/releases being published. E.g. `whiskers-v2.1.0` and `catwalk-v1.3.0` from the repository [catppuccin/toolbox](https://github.com/catppuccin/toolbox). I have been able to identify these tags/releases with my `renovate.json` shown below:

```json
{
  "$schema": "https://docs.renovatebot.com/renovate-schema.json",
  "extends": ["config:recommended"],
  "packageRules": [
    {
      "matchPackageNames": ["catppuccin/toolbox"],
      "versioning": "regex:^(?<packageName>\\w+-)?v(?<major>\\d+)\\.(?<minor>\\d+)\\.(?<patch>\\d+)?$"
    }
  ]
}
```

While I understand that the default `packageName` is the repository itself, I would like to change it to the actual binary, e.g. `whiskers` and `catwalk`. The repository name is confusing as branches, issues and PRs are created underneath the repository name and not the actual package name:

E.g.

`Dependency Dashboard`
![image](https://github.com/renovatebot/renovate/assets/58985301/991bf477-2479-4c42-ad3c-14cc93b9d6cf)

`Git Branches`
![image](https://github.com/renovatebot/renovate/assets/58985301/b9ec4e5d-1e6d-43f8-a1a4-c3d81412a50c)

`PRs`
![image](https://github.com/renovatebot/renovate/assets/58985301/4fbbf26e-3537-4b35-bee0-5f01f049ab1e)

#### Question

How can I retrieve the `packageName` from my custom versioning scheme and align it with the `whiskers` and `catwalk` binaries? Thanks!

**Minimal Reproduction Repository:** [sgoudham/renovate-homebrew-minimal-reproduction](https://github.com/sgoudham/renovate-homebrew-minimal-reproduction)

**P.S** I've also noticed that `v` is also incorrectly prefixed to the `whiskers` binary. I'd also appreciate help on how to fix that but happy to raise a separate discussion if that is preferred.

### Logs (if relevant)

<details><summary>Logs</summary>

```
DEBUG: branchifyUpgrades
DEBUG: detectSemanticCommits()
DEBUG: semanticCommits: returning "enabled" from cache
DEBUG: 3 flattened updates found: catppuccin/toolbox, catppuccin/toolbox, catppuccin/toolbox
DEBUG: Returning 2 branch(es)
DEBUG: config.repoIsOnboarded=true
DEBUG: packageFiles with updates
{
  "baseBranch": "main"
  "config": {
    "homebrew": [
      {
        "deps": [
          {
            "currentValue": "catwalk-v1.3.0",
            "currentVersion": "catwalk-v1.3.0",
            "currentVersionTimestamp": "2024-04-27T11:07:18.000Z",
            "datasource": "github-tags",
            "depName": "catppuccin/toolbox",
            "fixedVersion": "catwalk-v1.3.0",
            "isSingleVersion": true,
            "managerData": {
              "ownerName": "catppuccin",
              "repoName": "toolbox",
              "sha256": "e660ff504641d0eec9a69c2945970cfd121de9c7e6dd769c6ba3a72fed76700a",
              "url": "https://github.com/catppuccin/toolbox/archive/refs/tags/catwalk-v1.3.0.tar.gz"
            },
            "packageName": "catppuccin/toolbox",
            "registryUrl": "https://github.com",
            "sourceUrl": "https://github.com/catppuccin/toolbox",
            "versioning": "regex:^(?<packageName>\\w+-)?v(?<major>\\d+)\\.(?<minor>\\d+)\\.(?<patch>\\d+)?$",
            "warnings": [],
            "updates": [
              {
                "bucket": "non-major",
                "newVersion": "catwalk-v1.3.1",
                "newValue": "catwalk-v1.3.1",
                "releaseTimestamp": "2024-05-20T22:23:10.000Z",
                "newMajor": 1,
                "newMinor": 3,
                "updateType": "patch",
                "branchName": "renovate/catppuccin-toolbox-1.x"
              },
              {
                "bucket": "major",
                "newVersion": "whiskers-v2.1.1",
                "newValue": "whiskers-v2.1.1",
                "releaseTimestamp": "2024-05-20T22:23:10.000Z",
                "newMajor": 2,
                "newMinor": 1,
                "updateType": "major",
                "branchName": "renovate/catppuccin-toolbox-2.x"
              }
            ]
          }
        ],
        "packageFile": "Formula/catwalk.rb"
      },
      {
        "deps": [
          {
            "currentValue": "whiskers-v2.1.0",
            "currentVersion": "whiskers-v2.1.0",
            "currentVersionTimestamp": "2024-04-27T11:07:18.000Z",
            "datasource": "github-tags",
            "depName": "catppuccin/toolbox",
            "fixedVersion": "whiskers-v2.1.0",
            "isSingleVersion": true,
            "managerData": {
              "ownerName": "catppuccin",
              "repoName": "toolbox",
              "sha256": "5a3da3f7453ffc8929b6c5f6a3e925fa7bec0eb072e6b3ddaf1dcc511c464ae9",
              "url": "https://github.com/catppuccin/toolbox/archive/refs/tags/whiskers-v2.1.0.tar.gz"
            },
            "packageName": "catppuccin/toolbox",
            "registryUrl": "https://github.com",
            "sourceUrl": "https://github.com/catppuccin/toolbox",
            "versioning": "regex:^(?<packageName>\\w+-)?v(?<major>\\d+)\\.(?<minor>\\d+)\\.(?<patch>\\d+)?$",
            "warnings": [],
            "updates": [
              {
                "bucket": "non-major",
                "newVersion": "whiskers-v2.1.1",
                "newValue": "whiskers-v2.1.1",
                "releaseTimestamp": "2024-05-20T22:23:10.000Z",
                "newMajor": 2,
                "newMinor": 1,
                "updateType": "patch",
                "branchName": "renovate/catppuccin-toolbox-2.x"
              }
            ]
          }
        ],
        "packageFile": "Formula/whiskers.rb"
      }
    ]
  }
}

DEBUG: detectSemanticCommits()
DEBUG: semanticCommits: returning "enabled" from cache
DEBUG: processRepo()
DEBUG: Processing 2 branches: renovate/catppuccin-toolbox-1.x, renovate/catppuccin-toolbox-2.x
```

</details>

**Comment by rarkins** [comment]
It sounds like you should be using `depName` in addition to `packageName`. The former is the "pretty" name, and can be anything. `packageName` is what's used for lookups, and needs to be accurate for the lookup to succeed.

**Reply by sgoudham** [reply]
Hi 👋 

I tried using `depName` in a [future commit](https://github.com/sgoudham/renovate-homebrew-minimal-reproduction/commit/aeaad43e62fa03af1f242a23bc9c458ec26d075f) but the value wasn't updated in the logs and I was seeing the same incorrect behaviour.

Could you please advise on how to specify the `packageName` and `depName` while using the built-in homebrew module? My minimal reproduction repository is still up -> [sgoudham/renovate-homebrew-minimal-reproduction](https://github.com/sgoudham/renovate-homebrew-minimal-reproduction) with the [renovate.json](https://github.com/sgoudham/renovate-homebrew-minimal-reproduction/blob/main/renovate.json).

Thank you!

**Reply by rarkins** [reply]
Please update the reproduction describing expected vs observed behavior. When I look at the repo, I'm not certain what you're trying to demonstrate

**Reply by sgoudham** [reply]
Hi @rarkins 

I've updated my minimal reproduction repository, outlining the [actual behaviour](https://github.com/sgoudham/renovate-homebrew-minimal-reproduction#actual-behaviour) and my [expected behaviour](https://github.com/sgoudham/renovate-homebrew-minimal-reproduction#expected-behaviour). I hope this clarifies my situation, please take a look at your earliest convenience!

Thanks!

**Reply by rarkins** [reply]
Thanks. Based on how the homebrew manager works, I'm not sure this is achievable without you using groupName.

The key question is, should we have designed the manager so that the class name is "depName"?

```
class Whiskers < Formula
  desc "Soothing port creation tool for the high-spirited!"
  homepage "https://github.com/catppuccin/toolbox/tree/main/whiskers"
  url "https://github.com/catppuccin/toolbox/archive/refs/tags/whiskers-v2.1.0.tar.gz"
```

e.g. if users get a PR for the above, would they expect Renovate to say "Update Whiskers to v2.1.0" or "Update catppuccin/toolbox to v2.1.0"? In your case you want Whiskers, but would you want that for every single class by default, and would everyone else?

The homebrew manager was added long ago: https://github.com/renovatebot/renovate/pull/3530

We maybe didn't even have the possibility then to separate depName vs packageName
